llms/mlx_lm/generate.py

```diff
@@ -35,15 +35,31 @@ def setup_arg_parser():
         "--temp", type=float, default=DEFAULT_TEMP, help="Sampling temperature"
     )
     parser.add_argument("--seed", type=int, default=DEFAULT_SEED, help="PRNG seed")
+    parser.add_argument(
+        "--apply_chat_template",
+        "-a",
+        action="store_true",
+        help="apply the tokenizer's chat template on the prompt",
+    )
     return parser
 
 
 def main(args):
     mx.random.seed(args.seed)
     model, tokenizer = load(args.model)
+    if args.apply_chat_template:
+        messages = [{"role": "user", "content": args.prompt}]
+        # test if the tokenizer has a chat template
+        if not hasattr(tokenizer, "apply_chat_template"):
```

Comment:
nit: could add one more check if `tokenizer.chat_template` is set. Otherwise, `apply_chat_template` may use the default template, which can cause some unexpected behavior. https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils_base.py#L1732-L1737

Reply:
I've now added a message to warn about that , owever while testing it I see that the transformers lib also prints a warning, so it may be too much? Thank you for your reviews and tell me what you think.

example of missing template:
```python
❯ python -m mlx_lm.generate --model TinyLlama/TinyLlama-1.1B-Chat-v0.3 --max-tokens 20 --prompt 'I have a capybara in my garden! What should I do?' -a
Fetching 8 files: 100%|███████████████████████| 8/8 [00:00<00:00, 122910.01it/s]
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
TinyLlama/TinyLlama-1.1B-Chat-v0.3 does not define a chat template, using the default template, which may cause unexpected results

No chat template is defined for this tokenizer - using the default template for the LlamaTokenizerFast class. If the default is not appropriate for your model, please set `tokenizer.chat_template` to an appropriate template. See https://huggingface.co/docs/transformers/main/chat_templating for more information.

==========
Prompt: (...)
```

(I've also rework the other error message to better reflect the problem I think)

Reply:
Maybe only enable the chat template when the tokenizer has applied a chat template and has a tokenizer.chat_template. 
For example:
```python
if hasattr(tokenizer, "apply_chat_template") and tokenizer.chat_template:
    prompt = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
```

Reply:
I like this idea, let's use the prompt as is when no chat template exists. That seems clean to me.